rskj-core/src/main/java/co/rsk/peg/federation/FederationSupport.java

```diff
@@ -0,0 +1,48 @@
+package co.rsk.peg.federation;
+
+import co.rsk.bitcoinj.core.Address;
+import co.rsk.bitcoinj.core.UTXO;
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.peg.vote.ABICallSpec;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+
+import javax.annotation.Nullable;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+
+public interface FederationSupport {
+
+    Federation getActiveFederation();
+    Optional<Script> getActiveFederationRedeemScript();
+    Address getActiveFederationAddress();
+    int getActiveFederationSize();
+    Integer getActiveFederationThreshold();
```

Comment:
I notice some of the return types are primitives while others are wrappers. Do you think it would be possible to use the primitive type always? That way we avoid doing null checks and such

Reply:
very good point actually. @marcos-iov you want me to create a task to take care of this? Or should I just do it here?

Reply:
You can probably simply do it here. You'll probably need to do some adjustments when this gets integrated but that will be on different task

Reply:
Perfect, thanks !!!!

Reply:
done